Device not detected when recently reset with the latest luniistore version - Could not locate device partition

**Describe the bug**
A recently reset device (lunii v2) is not detected by Studio. The GUI doesn't show any error and the logs show a "Could not locate device partition" stacktrace (see more below)

**I have a second device available (lunii v1) that is still correctly detected.** So this is not a configuration issue.

The device that is not detected **was still detected two weeks ago.** In the meantime:
- I reset that device
- I upgraded my OS version


**To Reproduce**
Steps to reproduce the behavior:
1. Go to lunii store, reset a device
2. Quit store, restart device, restart computer
3. Start studio
4. See that the device is not detected

**Expected behavior**
The device should be detected


**Logs**
```

2022-11-17 13:56:20,043 [INFO ] [s.w.MainVerticle]: Opening URL in default browser...
2022-11-17 13:56:20,376 [INFO ] [i.v.c.i.l.c.VertxIsolatedDeployer]: Succeeded in deploying verticle
2022-11-17 13:56:30,187 [ERROR] [s.d.LibUsbDetectionHelper]: An error occurred while handling device plug event
java.util.concurrent.CompletionException: studio.driver.StoryTellerException: Could not locate device partition
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1739)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1728)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: studio.driver.StoryTellerException: Could not locate device partition
	at studio.driver.fs.FsStoryTellerAsyncDriver$1.onDevicePlugged(FsStoryTellerAsyncDriver.java:78)
	at studio.driver.LibUsbDetectionHelper.lambda$registerCallback$1(LibUsbDetectionHelper.java:118)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1736)
	... 6 more
```

**Desktop and environment (please complete the following information):**
 - OS:  mac osX 13.0.1
 - Browser: all
 - Java version: zulu 11
 - Maven version: 3.8.4
 - STUdio application version 0.3.1

**Additional context**
I can see a difference between my two devices at the partitioning level. I don't know if this difference is due a recent change in luuni reset procedure or if it is due to being from two hardware with different versions (V1 vs V2)

The failing one is the first
```
/dev/disk4 (external, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:     FDisk_partition_scheme                        *7.9 GB     disk4
   1:                 DOS_FAT_32 LUNII                   7.9 GB     disk4s1

/dev/disk6 (external, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:                            LUNII                  *7.9 GB     disk6

```

i put this line with textedit ... Sorry i'm so a begginer 

i still have this : 

2022-11-30 22:20:51,521 [DEBUG] [s.d.LibUsbAsyncEventsWorker]: Starting worker thread to handle libusb async events...
2022-11-30 22:20:51,521 [INFO ] [s.d.LibUsbDetectionHelper]: Hotplug event callback (0c45:6840): 1
2022-11-30 22:20:51,524 [DEBUG] [s.d.f.FsStoryTellerAsyncDriver]: Waiting for device partition...
2022-11-30 22:20:51,604 [INFO ] [s.w.MainVerticle]: Opening URL in default browser...
2022-11-30 22:20:51,822 [INFO ] [i.v.c.i.l.c.VertxIsolatedDeployer]: Succeeded in deploying verticle
2022-11-30 22:21:01,630 [ERROR] [s.d.LibUsbDetectionHelper]: An error occurred while handling device plug event
java.util.concurrent.CompletionException: studio.driver.StoryTellerException: Could not locate device partition
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1803)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1792)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:295)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1016)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1665)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1598)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: studio.driver.StoryTellerException: Could not locate device partition
	at studio.driver.fs.FsStoryTellerAsyncDriver$1.onDevicePlugged(FsStoryTellerAsyncDriver.java:78)
	at studio.driver.LibUsbDetectionHelper.lambda$registerCallback$1(LibUsbDetectionHelper.java:118)
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1800)
	... 6 more

--- a/driver/src/main/java/studio/driver/fs/DeviceUtils.java
+++ b/driver/src/main/java/studio/driver/fs/DeviceUtils.java
@@ -32,8 +32,8 @@ public class DeviceUtils {
                     .map(root -> root.toPath().toString())
                     .collect(Collectors.toList());
         } else {
-            final String CMD_DF = "df -l";
-            final Pattern dfPattern = Pattern.compile("^(\\/[^ ]+)[^/]+(/.*)$");
+            final String CMD_DF = SystemUtils.IS_OS_MAC ? "df" : "df -l";
+            final Pattern dfPattern = Pattern.compile("^([^ ]+)[^/]+(/.*)$");
 
             List<String> mountPoints = new ArrayList<>();
 
@@ -51,7 +51,7 @@ public class DeviceUtils {
                     if (matcher.matches()) {
                         final String dev = matcher.group(1);
                         final String rootPath = matcher.group(2);
-                        if (dev.startsWith("/dev/")) {
+                        if (dev.startsWith("/dev/") || SystemUtils.IS_OS_MAC) {
                             mountPoints.add(rootPath);
                         }
                     }

